### PR TITLE
External sensor support to compass HMC5883L added

### LINKED ISSUE #1288: No support for external sensor in Compass instrument
**Actual Behaviour**

Only built-in sensors can be used in Compass instrument

**Expected Behaviour**

An external sensor like HMC5883L should also be supported in Compass instrument

**Steps to reproduce it**

Observe Compass instrument

**LogCat for the issue**

Not applicable

**Screenshots of the issue**

Not applicable

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by ho-dor**
Fixes #1288 

**Changes**: External sensor support for compass.

**Screenshot/s for the changes**: 

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**:
[compass.zip](https://github.com/fossasia/pslab-android/files/3342882/compass.zip)

### CODE REVIEW

**Comment by ho-dor** [comment]
@neel1998 @CloudyPadmal please review.

File: app/src/main/java/io/pslab/activity/CompassActivity.java [hunk 1]
```diff
@@ -96,4 +99,18 @@ public void getDataFromDataLogger() {
...
+    protected void onResume() {
+        super.onResume();
+        reinstateConfigurations();
+    }
+
+    private void reinstateConfigurations() {
+        SharedPreferences compassConfigurations;
+        compassConfigurations = PreferenceManager.getDefaultSharedPreferences(getBaseContext());
+        locationEnabled = compassConfigurations.getBoolean(ThermometerSettingsFragment.KEY_INCLUDE_LOCATION, true);
+        CompassDataFragment.setParameters(
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/activity/CompassActivity.java]: What's a thermo setting doing in here?

File: app/src/main/java/io/pslab/fragment/CompassDataFragment.java [hunk 2]
```diff
@@ -597,13 +606,47 @@ private void resetInstrumentData() {
...
+    private void initiateCompassSensor(int type) {
+
+        CompassDataFragment.COMPASS_SENSOR s = CompassDataFragment.COMPASS_SENSOR.values()[type];
+        resetInstrumentData();
+        ScienceLab scienceLab;
+        switch (s) {
+            case INBUILT_SENSOR:
+                degreeIndicator.setText(getResources().getStringArray(R.array.compass_sensors)[0]);
+                sensorManager = (SensorManager) getContext().getSystemService(SENSOR_SERVICE);
+                sensor = sensorManager.getDefaultSensor(Sensor.TYPE_AMBIENT_TEMPERATURE);
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/fragment/CompassDataFragment.java]: These are mixed up somehow. Did you update the branch correctly?

**Comment by cweitat** [comment]
Your GIF shows the word compass configurations without spacing in between.. can you check?

**Comment by CloudyPadmal** [comment]
> Your GIF shows the word compass configurations without spacing in between.. can you check?

May be the sentence is too long :thinking: 
Here it got a space: https://github.com/fossasia/pslab-android/pull/1841/files#diff-a13d0d8be0dcb8b9698023380937a6ebR103

File: app/src/main/java/io/pslab/fragment/CompassDataFragment.java (hunk 2 above)
Comment (CloudyPadmal) [reply app/src/main/java/io/pslab/fragment/CompassDataFragment.java]: These parts not resolved yet

**Comment by cweitat** [comment]
> > Your GIF shows the word compass configurations without spacing in between.. can you check?
> 
> May be the sentence is too long 🤔
> Here it got a space: https://github.com/fossasia/pslab-android/pull/1841/files#diff-a13d0d8be0dcb8b9698023380937a6ebR103

Yup I checked the code was fine. So was unsure ;)

**Comment by ho-dor** [comment]
@CloudyPadmal I have made the changes

**Pull request merged**